File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day19MedicineForRudolphTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day19MedicineForRudolphTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day19MedicineForRudolph("/twenty_fifteen/Day19-MedicineForRudolph-input.txt");

        assertEquals(518, dp.getPart1Answer().intValue());
        assertEquals(200, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day19MedicineForRudolph("/twenty_fifteen/Day19-MedicineForRudolph-example#1.txt");

        assertEquals(4, dp.getPart1Answer().intValue());
        assertEquals(3, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day19MedicineForRudolph("/twenty_fifteen/Day19-MedicineForRudolph-example#2.txt");

        assertEquals(7, dp.getPart1Answer().intValue());
        assertEquals(6, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day19MedicineForRudolphTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day19MedicineForRudolphTest  extends FlatSpec with Matchers {
  "2015 Day 19 - Input" should "calculate answers" in {
    val dp = new Day19MedicineForRudolph("/twenty_fifteen/Day19-MedicineForRudolph-input.txt")
    dp.part1Answer should be(518)
    dp.part2Answer should be(200)
  }

  "2015 Day 19 - Example #1" should "calculate answers" in {
    val dp = new Day19MedicineForRudolph("/twenty_fifteen/Day19-MedicineForRudolph-example#1.txt")
    dp.part1Answer should be(4)
    dp.part2Answer should be(3)
  }

  "2015 Day 19 - Example #2" should "calculate answers" in {
    val dp = new Day19MedicineForRudolph("/twenty_fifteen/Day19-MedicineForRudolph-example#2.txt")
    dp.part1Answer should be(7)
    dp.part2Answer should be(6)
  }
}
